instrumentation/sequel/lib/sequel/extensions/opentelemetry.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'sequel'
+
+# Sequel: The Database Toolkit for Ruby
+module Sequel
+  # The Sequel::Opentelemetry extension
+  #
+  # @example Simple usage
+  #
+  #     require 'sequel-opentelemetry'
+  #     DB.extension :opentelemetry
+  #
+  # Related module: Sequel::Opentelemetry
+  module Opentelemetry
+    def tracer
+      @tracer ||= ::OpenTelemetry.tracer_provider.tracer('Sequel', ::Sequel::VERSION)
+    end
+
+    def trace_execute(super_method, sequel_method, sql, options, &block)
+      response = nil
+      attributes = {
+        'component' => 'Sequel',
+        ::OpenTelemetry::SemanticConventions::Trace::DB_NAME => opts[:database],
+        ::OpenTelemetry::SemanticConventions::Trace::DB_SYSTEM => database_type.to_s,
+        ::OpenTelemetry::SemanticConventions::Trace::DB_STATEMENT => sql,
+        ::OpenTelemetry::SemanticConventions::Trace::DB_OPERATION => sequel_method
+      }
+      attributes['sequel.host'] = opts[:host] if opts[:host]
+
+      tracer.in_span("sequel.#{sequel_method}", attributes: attributes) do
+        response = super_method.call(sql, options, &block)
+      end
+      response
+    end
+
+    def execute(sql, options = ::Sequel::OPTS, &block)
+      trace_execute(proc { super(sql, options, &block) }, 'execute', sql, options, &block)
+    end
+
+    def execute_dui(sql, options = ::Sequel::OPTS, &block)
+      trace_execute(proc { super(sql, options, &block) }, 'execute_dui', sql, options, &block)
+    end
+  end
+  Database.register_extension(:opentelemetry, Opentelemetry)
```

Comment:
I this best done here or as part of the `instrumentation.install` method?
```suggestion
  Database.register_extension(:opentelemetry, self)
```